http://gcc.gnu.org/PR35705

2008-04-01  John David Anglin  <dave.anglin@nrc-cnrc.gc.ca>

	PR middle-end/35705
	* fold-const.c (get_pointer_modulus_and_residue): Return modulus 1 if
	the expression is a function address.

--- branches/gcc-4_3-branch/gcc/fold-const.c	2008/04/01 22:14:41	133804
+++ branches/gcc-4_3-branch/gcc/fold-const.c	2008/04/01 22:16:49	133805
@@ -9360,7 +9360,7 @@
 	    }
 	}
 
-      if (DECL_P (expr))
+      if (DECL_P (expr) && TREE_CODE (expr) != FUNCTION_DECL)
 	return DECL_ALIGN_UNIT (expr);
     }
   else if (code == POINTER_PLUS_EXPR)
